src/outputs/parthenon_hdf5.cpp

```diff
@@ -245,28 +245,25 @@ void PHDF5Output::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) {
   /* create an MPI_INFO object -- on some platforms it is useful to
      pass some information onto the underlying MPI_File_open call */
   MPI_Info FILE_INFO_TEMPLATE;
-  int ierr;
-  MPI_Status stat;
-  ierr = MPI_Info_create(&FILE_INFO_TEMPLATE);
-  ierr = H5Pset_sieve_buf_size(acc_file, 262144);
-  ierr = H5Pset_alignment(acc_file, 524288, 262144);
+  MPI_Info_create(&FILE_INFO_TEMPLATE);
+  H5Pset_sieve_buf_size(acc_file, 262144);
+  H5Pset_alignment(acc_file, 524288, 262144);
```

Comment:
I don't like removing `ierr` here... I understand why you did it, to remove the warning. But instead, I think we should check `ierr` and throw if it's not `H5_SUCCESS`

(And same elsewhere where it's been removed, obviously.)

Reply:
I also think it's worth keeping this.
Could we may just to a `ierr &= ...` and do a final check or similar?

Reply:
I think it should be `+=`, but yes, I like checking it at the end.

Reply:
I'm adding macros to `error_checking.hpp` and checking each function call... I don't like the accumulation idea because it means you continue to execute additional library functions after you've entered a presumably erroneous state.